src/shogun/distance/EuclideanDistance.h

```diff
@@ -125,6 +161,21 @@ class CEuclideanDistance: public CRealDistance
 	protected:
 		/** if application of sqrt on matrix computation is disabled */
 		bool disable_sqrt;
+
+		/** if squared norms of rhs features are used */
+		bool m_rhs_sq_norms;
+
+		/** if squared norms of lhs features are used */
+		bool m_lhs_sq_norms;
+
+		/** i dot products are used in distance computations */
+		bool m_dot_enabled;
+
+		/** squared norms from features of right hand side to be used if dot enabled */
+		SGVector<float64_t> m_rhs_squared_norms;
```

Comment:
We are not calculating the whole distance matrix. It is just between 2 vectors from lhs and rhs  `distance (index a, index b)`
So if we know that either lhs or rhs or both are fixed over some number of computations we can just use `rhs_norm[ index b]` instead of computing `b.b` for every computation. That was the idea.